### PR TITLE
fix: gas estimator calculation

### PR DESCRIPTION
**Pull request by md0x**
**Motivation**

Explain the motivation for making this change.
What existing problem does the pull request solve?
It may be sufficient to just link to an issue, but please explain if more details are needed.

**Summary**

Briefly summarize what changes were made to accomplish the motivation above.

**Details**

This may be unnecessary for some PRs. Catch-all for detailed explanations about the implementation decisions and implications of the change.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes #XXXX

### CODE REVIEW

File: packages/financial-templates-lib/src/helpers/GasEstimator.ts [hunk 1]
```diff
@@ -255,9 +255,8 @@ export class GasEstimator {
       if (etherscanGasResponse.result.suggestBaseFee === undefined)
         throw new Error(`Bad ethgasstation response ${json}`);
       return {
-        maxFeePerGas: Number(etherscanGasResponse.result.FastGasPrice),
-        maxPriorityFeePerGas:
-          Number(etherscanGasResponse.result.FastGasPrice) - Number(etherscanGasResponse.result.suggestBaseFee),
+        maxFeePerGas: Number(etherscanGasResponse.result.suggestBaseFee) * 2,
+        maxPriorityFeePerGas: Number(etherscanGasResponse.result.suggestBaseFee),
```
Comment (mrice32) [inline packages/financial-templates-lib/src/helpers/GasEstimator.ts:259]: This will be a very high priority fee, at least for mainnet (not as sure about polygon). What's the rationale for setting the priority fee to the suggested base fee?

Comment (md0x) [reply packages/financial-templates-lib/src/helpers/GasEstimator.ts:259]: @mrice32 I think we decided to do maxFeePerGas = 2 * base fee, actually I've checked and it was 3 times.

Given that `maxFeePerGas = baseFeePerGas + maxPriorityFeePerGas` it means that `maxPriorityFeePerGas` should be `baseFeePerGas`

Do you agree?

Comment (mrice32) [reply packages/financial-templates-lib/src/helpers/GasEstimator.ts:259]: I'm not sure I agree because of the maxes.

feePerGas = baseFeePerGas + priorityFeePerGas

The way I think about the maxes is that they put a cap on each of the values. We are allowed to cap the total fee and the priority fee. The difference between those caps is how high the base fee can get (but it can always be lower, meaning you'll pay less!).

Setting a high maxFeePerGas, but a fairly low maxPriorityFeePerGas just means that you are constraining the tip (tips rarely need to be higher than 1 gwei or so), but allowing the algorithmic base bee (burned amount) to fluctuate without blocking your transaction.

I would suggest either setting a hardcoded maxPriorityFeePerGas or using an API value that you are confident will not make it very high. High tips (maxPriorityFeePerGas) are basically just giving away money, but high maxFeePerGas (room for the base fee to fluctuate) is what stops your txn from getting blocked bc it is impossible to include.

TLDR: maxPriorityFeePerGas should always be less than 2 gwei (even values as low as 0.1 gwei are probably fine!). maxFeePerGas should be 2-3x current base fee.

Does that make sense?

**Review by mrice32 — COMMENTED** [review]
https://github.com/UMAprotocol/protocol/pull/4593#discussion_r1264005888

File: packages/financial-templates-lib/src/helpers/GasEstimator.ts (hunk 1 above)
Comment (md0x) [reply packages/financial-templates-lib/src/helpers/GasEstimator.ts]: Thanks @mrice32,
I've updated the calculations to set the `maxFeePerGas` to be 3 x the baseFee and hardcoded the `maxPriorityFeePerGas` in mainnet and polygon (5 gwei and 50gwei). I've decided the hardcoded values by looking at priority values showed here: https://etherscan.io/gastracker (3~6 gwei) and here https://polygonscan.com/gastracker (30~70 gwei)

File: packages/financial-templates-lib/src/helpers/GasEstimator.ts [hunk 2]
```diff
@@ -251,19 +251,19 @@ export class GasEstimator {
 
   private _extractFastGasPrice(json: { [key: string]: any }, url: string): LondonGasData | LegacyGasData {
     if (url.includes("etherscan.io") || url.includes("polygonscan.com")) {
+      const isMainnet = url.includes("etherscan.io");
```
Comment (mrice32) [inline packages/financial-templates-lib/src/helpers/GasEstimator.ts:254]: Will optimism cause problems here? Their url is optimistic.etherscan.io. Or do we need to handle optimism at all?

File: packages/financial-templates-lib/src/helpers/GasEstimator.ts [hunk 3]
```diff
@@ -251,19 +251,19 @@ export class GasEstimator {
...
       const etherscanGasResponse = json as EtherscanGasResponse;
       if (etherscanGasResponse.result.suggestBaseFee === undefined)
         throw new Error(`Bad ethgasstation response ${json}`);
       return {
-        maxFeePerGas: Number(etherscanGasResponse.result.FastGasPrice),
-        maxPriorityFeePerGas:
-          Number(etherscanGasResponse.result.FastGasPrice) - Number(etherscanGasResponse.result.suggestBaseFee),
+        maxFeePerGas: Number(etherscanGasResponse.result.suggestBaseFee) * 3,
+        maxPriorityFeePerGas: isMainnet ? 5 : 50,
```
Comment (mrice32) [inline packages/financial-templates-lib/src/helpers/GasEstimator.ts:260]: 5 is really high for priority fee. A low priority fee will _only not be included_ if the block is 100% full of txns with a higher priority fee (which causes the base fee to increase). I would set this to 1.

For polygon, they apparently don't implement EIP 1559 "correctly", so it does need to be set fairly high. 50 is probably fine, but I've read that you really do need to use a gas estimating service to really get a good number. Since we probably don't send a lot of polygon transfers from these bots, overpricing on polygon is probably fine.

**Review by mrice32 — APPROVED** [review]
Looks great! Just a few minor comments, but overall, I really like this change!

File: packages/financial-templates-lib/src/helpers/GasEstimator.ts (hunk 2 above)
Comment (md0x) [reply packages/financial-templates-lib/src/helpers/GasEstimator.ts:254]: [Optimism is just defaulted](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/helpers/GasEstimator.ts#L209)

Also I don't think the GasEstimator is used there

File: packages/financial-templates-lib/src/helpers/GasEstimator.ts (hunk 3 above)
Comment (md0x) [reply packages/financial-templates-lib/src/helpers/GasEstimator.ts]: Thanks @mrice32, set mainnet maxPriorityFeePerGas to 1 in latest commit
